Add Signed Request Object section to CAMARA Security and Interoperability Profile (#233)
camaraproject/IdentityAndConsentManagement: documentation/CAMARA-Security-Interoperability.md

The OIDC Authorization Code Flow is defined in [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html)

### Signed Request Object

> It is RECOMMENDED to use signed Request Object, a [JSON Web Token (JWT) \[RFC7519\]](https://www.rfc-editor.org/rfc/rfc7519.html), passed by value, as specified by [OpenID Connect Core](https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests) and [\[RFC9101\]](https://www.rfc-editor.org/rfc/rfc9101.html#name-request-object-2).

Reviewer: ```suggestion
It is RECOMMENDED to use a signed request object as specified by [OpenID Connect Core](https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests) [pass the signed request object by value](https://openid.net/specs/openid-connect-core-1_0.html#RequestObject).
```

> The `sub` (Subject) claim in the signed Request Object MUST NOT be present, to prevent [cross-JWT confusion](https://www.rfc-editor.org/rfc/rfc9101.html#name-cross-jwt-confusion).

Reviewer: ```suggestion
```

`sub` is not part of https://openid.net/specs/openid-connect-core-1_0.html#RequestObject

> The value of `aud` (Audience) claim MUST be the Authorization endpoint as RECOMMENDED by [\[RFC9101\]](https://www.rfc-editor.org/rfc/rfc9101.html#name-explicit-endpoints).

Reviewer: ```suggestion
The value of `aud` (Audience) claim MUST be set to a value that the authorization server accepts as itself.
This document recommends in the [client credentials section](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md#client-authentication) that the `aud` value SHOULD be the URL of the Authorization Server's [Token Endpoint](https://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint). The same recommendation is given for the `aud` value of the signed request object.
```

> The following query string parameters of the authorization request URI are REQUIRED, the rest of the [request parameters](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) are OPTIONAL:

Reviewer: ```suggestion
```

Does this section change what is defined in OIDC? 
The request parameter cannot be REQUIRED if the signed request object is RECOMMENDED.
Can this section be removed because no change to OIDC?
